nixos/modules/security/acme.nix

```diff
@@ -268,6 +269,8 @@ let
             ${data.postRun}
           fi
         '');
+      } // optionalAttrs (!useDns && data.webroot == null) {
+        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
```

Comment:
Any idea, why you can set `--http.port` for Lego? Does ACME work on other ports than `80/tcp`? If so, this could be conditional on a port option being below 1024.

Reply:
> Our implementation of the HTTP-01 challenge follows redirects, up to 10 redirects deep. It only accepts redirects to “http:” or “https:”, and only to ports 80 or 443.

(https://letsencrypt.org/docs/challenge-types/)

So it's basically reverse proxying lego (requires local networking) or serving a static directory (requires file/directory permissions), got it. Guess we should expose the http.port as an option then, given that it would default to `:80` and that likely wouldn't work with other webservers in place.

```
    --http.port value            Set the port and interface to use for HTTP based challenges to listen on.Supported: interface:port or :port. (default: ":80")
```

Reply:
Sounds reasonable. What about `--http.proxy-header`? I'm not super familiar with proxying, so I don't know if this is necessary:

> Validate against this HTTP header when solving HTTP based challenges behind a reverse proxy. (default: "Host")


Additionally, we should probably try to make the challenge configuration a bit less brittle in the future, given that they're not mutually exclusive:

```
   --http                       Use the HTTP challenge to solve challenges. Can be mixed with other types of challenges.
   --tls                        Use the TLS challenge to solve challenges. Can be mixed with other types of challenges.
   --dns value                  Solve a DNS challenge using the specified provider. Can be mixed with other types of challenges. Run 'lego dnshelp' for help on usage.
```

(but that's something that belongs in another issue)

Reply:
(It's conditional now)